File: frontend/shared/types/tournament.ts
import type { EventCategory, TournamentLevel, TournamentStatus } from './enums'

export interface Tournament {
  id: number
  name: string
  season_id: number
  event_category: EventCategory
  level: TournamentLevel
  group_name?: string
  location?: string
  tournament_date?: string
  status: TournamentStatus
  created_at: string
  updated_at: string
}

export interface TournamentCreate {
  name: string
  event_category: EventCategory
  level: TournamentLevel
  group_name?: string
  location?: string
  tournament_date?: string
}

export interface TournamentUpdate {
  name?: string
  event_category?: EventCategory
  level?: TournamentLevel
  group_name?: string
  location?: string
  tournament_date?: string
}

File: backend/app/schemas/tournament.py
from datetime import date, datetime

from pydantic import BaseModel


class TournamentCreate(BaseModel):
    name: str
    event_category: str
    level: str
    group_name: str | None = None
    start_date: date | None = None
    end_date: date | None = None
    location: str | None = None
    alumni_player_count: int | None = None


class TournamentUpdate(BaseModel):
    name: str | None = None
    event_category: str | None = None
    level: str | None = None
    group_name: str | None = None
    start_date: date | None = None
    end_date: date | None = None
    location: str | None = None
    alumni_player_count: int | None = None


class TournamentResponse(BaseModel):
    id: int
    season_id: int
    name: str
    event_category: str
    level: str
    group_name: str | None = None
    start_date: date | None = None
    end_date: date | None = None
    location: str | None = None
    status: str
    created_at: datetime | None = None
